src/librustc_codegen_llvm/type_of.rs

```diff
@@ -52,32 +52,35 @@ fn uncached_llvm_type<'a, 'tcx>(
     }
 
     let name = match layout.ty.kind {
-        ty::Closure(..) |
-        ty::Generator(..) |
-        ty::Adt(..) |
-        // FIXME(eddyb) producing readable type names for trait objects can result
+        ty::Closure(..) | ty::Generator(..) | ty::Adt(..) | ty::Foreign(..) | ty::Str
+            if cx.tcx.sess.opts.debugging_opts.fewer_names =>
```

Comment:
Could this use session `fewer_names()` method to retain type names when emitting LLVM IR?

Reply:
Oops, that's what I meant to use :facepalm:

Reply:
`cx.sess().fewer_names()` was my suggestion, no idea why this PR uses the flag directly.

Reply:
Fixed! I only briefly skimmed the link that @eddyb sent me earlier and it looked like it was just logic which changed the default of `fewer_names`.